packages/in_app_purchase/ios/Classes/FIAPProductRequestHandler.m

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FIAPProductRequestHandler.h"
+#import <StoreKit/StoreKit.h>
+
+#pragma mark - Main Handler
+
+@interface FIAPProductRequestHandler () <SKProductsRequestDelegate>
+
+@property(copy, nonatomic) ProductRequestCompletion completion;
+@property(strong, nonatomic) NSMutableSet *delegateObjects;
+@property(strong, nonatomic) SKProductsRequest *request;
+
+@end
+
+@implementation FIAPProductRequestHandler
+
+- (instancetype)initWithRequestRequest:(SKProductsRequest *)request {
+  self = [super init];
+  if (self) {
+    self.request = request;
+    request.delegate = self;
+  }
+  return self;
+}
+
+- (void)startWithCompletionHandler:(ProductRequestCompletion)completion {
+  self.completion = completion;
+  [self.request start];
+}
+
+- (void)productsRequest:(SKProductsRequest *)request
+     didReceiveResponse:(SKProductsResponse *)response {
+  if (self.completion) {
+    self.completion(response);
+  }
+  [self.delegate productRequestHandlerDidFinish:self];
+}
+
+@end
+
+#pragma mark - SKProduct Coders
+
+@implementation SKProduct (Coder)
+
+- (NSDictionary *)toMap {
+  NSMutableDictionary *map = [[NSMutableDictionary alloc] initWithDictionary:@{
+    @"localizedDescription" : self.localizedDescription ?: [NSNull null],
+    @"localizedTitle" : self.localizedTitle ?: [NSNull null],
+    @"productIdentifier" : self.productIdentifier ?: [NSNull null],
+    @"downloadable" : @(self.downloadable),
+    @"price" : self.price ?: [NSNull null],
+    @"downloadContentLengths" : self.downloadContentLengths ?: [NSNull null],
+    @"downloadContentVersion" : self.downloadContentVersion ?: [NSNull null]
+
+  }];
+  if (@available(iOS 10.0, *)) {
+    // TODO: NSLocle is a complex object, want to see the actual need of getting this expanded to
```

Comment:
TODOs should be mention an owner, and reference an issue, see example here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions